File: Java/프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.java
import java.util.*;

class Solution {
    public int solution(int n) {
        int temp;
        int mod =  1_000_000_007;
        int[] dp = new int[n+2];
        
        dp[0] = 1;
        dp[1] = 2;
        
        for (int i=2; i<n; i++) {
            temp = dp[i - 2] + dp[i - 1];
            dp[i] = temp >= mod ? temp % mod : temp;
        }
        return (dp[n - 1]);
    }
}

File: JavaScript/프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.js
function solution(n) {
    const mod = 1_000_000_007;
    var dp = [];
    
    dp.push(1);
    dp.push(2);
    
    for (let i=2; i<n; i++) {
        let temp = (dp[i - 1] + dp[i - 2]);
        if (temp >= mod) temp = temp % mod;
        dp.push(temp);
    }
    return dp[n - 1];
}